### PR TITLE
doc: Update install and testing guide 

### PR DESCRIPTION
**Pull request by Glutexo**
The installation [instructions](https://github.com/candlepin/subscription-manager/blob/a01dabd2b6f8f6078661b671c5431d601c8171ea/INSTALL.md) now support RHEL in addition to Fedora. The guide has been tested on RHEL 9.2.

There are small touches to the commands in [`INSTALL.md`](https://github.com/candlepin/subscription-manager/blob/a01dabd2b6f8f6078661b671c5431d601c8171ea/INSTALL.md) too:

* `PYTHONPATH` is not a shell command, it doesn’t require explicit `./` for the current directory.
* An alias pointing to the python executable in the virtual environment makes it work without having to activate it.

The testing [guide](https://github.com/candlepin/subscription-manager/blob/a01dabd2b6f8f6078661b671c5431d601c8171ea/TESTING.md) explicitly mentions requirements in the test-requirements.txt file and discourages from using `./setup.py test`.

Card IDs:

* [CCT-655](https://issues.redhat.com/browse/CCT-655)
* [CCT-656](https://issues.redhat.com/browse/CCT-656)

### CODE REVIEW

**Comment by Glutexo** [comment]
I checked the DNF manual page and `--setopt=install_weak_deps=False` is mentioned exaclty like this. Interesting.

**Comment by m-horky** [comment]
@Glutexo On Fedora, weak dependencies are

```shell
$ podman run -it --rm fedora:40 dnf install ...
...
Installing weak dependencies:
 cryptsetup-libs
 diffutils
 libfido2
 libxkbcommon
 python3-urllib3+socks
 qrencode-libs
 systemd-networkd
 systemd-resolved
 systemd-udev
```

**Comment by Glutexo** [comment]
Interesting! I wouldn’t expect that the dependencies would differ that much. 😮

**Comment by Glutexo** [comment]
Opening the pull request for review.

I remember that when I was preparing my subscription-manager environment for the first time, I needed to install `dbus-devel` too. This time, I was able to run both the `subscription-manager` itself and its tests without having it installed, so I didn’t add it to the requirements. It is possible that it was a part of my (unsuccessful) effort to run the tests with (deprecated) `./setup.py test`.

File: TESTING.md [hunk 1]
```diff
@@ -2,6 +2,15 @@
...
 
+First, install the tests dependencies from [`test-requirements.txt`](test-requirements.txt). Make sure your virtual environment is activated.
+
+```bash
+pip install -r test-requirements.txt
+```
+
+***NOTE:** Don’t use `./setup.py test`; the command will most probably fail. Running tests this way is deprecated and has been even removed from the most recent versions of setuptools.*
+
+
```
Comment (m-horky) [inline TESTING.md:13]: I don't think this is necessary, we're explicitly saying you should be running `pytest` just below.

File: INSTALL.md [hunk 2]
```diff
@@ -50,6 +58,9 @@ Other Fedora versions or distributions may require some adaptation.
    git clone https://github.com/candlepin/subscription-manager.git
    cd subscription-manager/
    ```
+   
+   ***NOTE**: For `./setup.py build_ext --inplace` to succeed, it is necessary to really clone the `subscription-manager` repository and not just upload the files. Otherwise, the command would fail on `git describe`.*
+
```
Comment (m-horky) [inline INSTALL.md:63]: Why would you do that? :)

This note should not be necessary, you just have to follow the instructions.

File: INSTALL.md [hunk 3]
```diff
@@ -9,23 +9,31 @@ To install subscription-manager, run
...
        python3-rpm python3-librepo libdnf-devel cmake
    ```
+   
+   On RHEL, you need to enable the [CodeReady Linux Builder](https://docs.redhat.com/en/documentation/red_hat_enterprise_linux/9/html/package_manifest/repositories#CodeReadyLinuxBuilder-repository) repository in order to gain access to the development packages like `libdnf-devel`.
 
+   ```bash
+   sudo subscription-manager repos --enable codeready-builder-for-rhel-9-x86_64-rpms
+   ```
+   
    <!-- libdnf-devel, cmake are required to build product-id plugin -->
```
Comment (m-horky) [inline INSTALL.md:32]: Please leave this comment directly under the `sudo dnf install` block.

File: INSTALL.md [hunk 4]
```diff
@@ -9,23 +9,31 @@ To install subscription-manager, run
 sudo dnf install subscription-manager
 ```
 
+This is not necessary on RHEL installations as subscription-manager is pre-installed and registration is required to install any packages with DNF.
+
```
Comment (m-horky) [inline INSTALL.md:13]: While true, I am not totally sure this is adding any value. Instead, maybe we could just say that

```md
RHEL already comes with `subscription-manager` pre-installed.
```

if we want to mention RHEL.

> registration is required to install any packages with DNF.

suggestion:

````md
With `subscription-manager` present, you can get register the system with your Red Hat account by running

```bash
sudo subscription-manager register
```
````

File: INSTALL.md [hunk 5]
```diff
@@ -9,23 +9,31 @@ To install subscription-manager, run
...
 ## Developer installation
 
-The process below has been tested on Fedora 36.
-Other Fedora versions or distributions may require some adaptation.
+The process below has been tested on Fedora 36 and RHEL 9.
+Other Fedora/RHEL versions or distributions may require some adaptation.
```
Comment (m-horky) [inline INSTALL.md:17]: Let's just drop the `Fedora/RHEL`, at this point it's clear.

**Comment by Glutexo** [comment]
Added JIRA card IDs to the commit message and to the description.

File: INSTALL.md [hunk 6]
```diff
@@ -71,7 +82,7 @@ Other Fedora versions or distributions may require some adaptation.
    ```bash
    alias subscription-manager="sudo \
        PYTHONPATH=/path/to/subscription-manager/src \
-       $(which python3) \
+       ~/.venvs/subscription-manager/bin/python3 \
```
Comment (m-horky) [inline INSTALL.md:85]: I wouldn't do these, it's up to everyone to rewrite the path to something that matches their system.

File: INSTALL.md [hunk 7]
```diff
@@ -82,11 +93,11 @@ Other Fedora versions or distributions may require some adaptation.
    ```bash
    alias rhsm-service="sudo \
        PYTHONPATH=/path/to/subscription-manager/src \
-       $(which python3) \
+       ~/.venvs/subscription-manager/bin/python3 \
        -m subscription_manager.scripts.rhsm_service --verbose"
    alias rhsmcertd="sudo \
        PYTHONPATH=/path/to/subscription-manager/src \
-       $(which python3) \
+       ~/.venvs/subscription-manager/bin/python3 \
```
Comment (m-horky) [inline INSTALL.md:100]: ditto

**Review by m-horky — CHANGES_REQUESTED** [review]

File: INSTALL.md (hunk 2 above)
Comment (Glutexo) [reply INSTALL.md:63]: The scenario is you have the repository cloned on the desktop you use to write code and uploaded to a (virtual) machine with RHEL to actually run the software or its test.

Comment (m-horky) [reply INSTALL.md:63]: Oh, okay. Still, I believe you should configure your VM to mount the directories from the host, so you don't have to synchronize it.

While your use-case is valid, it won't scale well :)

File: TESTING.md (hunk 1 above)
Comment (Glutexo) [reply TESTING.md:13]: Removed.

---

It may be still useful to mention that a standard way of running tests doesn’t work, just as anybody can be tempted to try runnin `make test` in a project with a Makefile. `setup.py` contains `test_require`, which is only used by `./setup.py test`, creating an impression this way of running tests is supported, or at least that it’s the way to install the test dependencies.

That’s how I found it, because `test-requirements.txt` is not mentioned in this guide and I didn’t notice the test requirements file. Searching for the name of the missing package lead me to the list in `setup.py` and from there to `tests_require`. 

Adding `pip install -r test-requirements.txt` to the guide can hopefully mitigate this. If we later remove `test_require` entirely, the space for confusion will disappear.

File: INSTALL.md (hunk 3 above)
Comment (Glutexo) [reply INSTALL.md:32]: Makes sense! Moved.

File: INSTALL.md (hunk 6 above)
Comment (Glutexo) [reply INSTALL.md:85]: This is the path we suggest by

```bash
mkdir -p ~/.venvs/
python3 -m venv --system-site-packages ~/.venvs/subscription-manager
```

The proble with `$(which python3)` is that it wouldn’t work without activating the virtual environment. Like that, it would make more sense to add the alias to the venv activation script rather than adding a global one.

I’ll put the `$(which python3)` back.

File: INSTALL.md (hunk 5 above)
Comment (Glutexo) [reply INSTALL.md]: Dropped.

**Comment by Glutexo** [comment]
Addressed the review comments, thank you! Asking @m-horky for a fresh review.

**Comment by Glutexo** [comment]
Addressed the review comment, re-requesting a review.

File: TESTING.md (hunk 1 above)
Comment (m-horky) [reply TESTING.md]: The tool is just old and rusty on too many places :)

File: INSTALL.md [hunk 8]
```diff
@@ -82,7 +95,7 @@ Other Fedora versions or distributions may require some adaptation.
    ```bash
    alias rhsm-service="sudo \
        PYTHONPATH=/path/to/subscription-manager/src \
-       $(which python3) \
+       $(which python3)  \
```
Comment (m-horky) [inline INSTALL.md:98]: Extra space

File: INSTALL.md [hunk 9]
```diff
@@ -61,7 +74,7 @@ Other Fedora versions or distributions may require some adaptation.
 6. Test your local installation:
 
    ```bash
-   sudo PYTHONPATH=./src python3 -m subscription_manager.scripts.subscription_manager
+   sudo PYTHONPATH=src python3 -m subscription_manager.scripts.subscription_manager
```
Comment (m-horky) [inline INSTALL.md:77]: Why this change?

**Review by m-horky — CHANGES_REQUESTED** [review]
Looks good, just two very small changes I'd like you to make.

File: INSTALL.md (hunk 9 above)
Comment (Glutexo) [reply INSTALL.md:77]: The `./` does not bring any value, it’s implicit. Environment variables aren’t shell commands, where `./` is required to run an executable in the current directory.

File: INSTALL.md (hunk 8 above)
Comment (Glutexo) [reply INSTALL.md:98]: Oh. I spotted it myself, but didn’t commit and push it. 🤦🏻‍♂️

File: INSTALL.md (hunk 9 above)
Comment (Glutexo) [reply INSTALL.md]: Reverted.

**Comment by m-horky** [comment]
Please drop the `Merge branch 'main' into install` commit (you probably wanted to rebase instead?). LGTM otherwise.

**Comment by Glutexo** [comment]
I did not want to rebase. I clicked the “Update branch” in GitHub.

**Comment by Glutexo** [comment]
Rebased on the current state of _main_.

**Review by m-horky — APPROVED** [review]
LGTM
